Nanoleaf-wpf/Api/Effects/ScreenMirrorEffects/ScreenMirror.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Winleafs.Api;
+using Winleafs.Api.Endpoints.Interfaces;
+using Winleafs.Wpf.Api.Layouts;
+using Winleafs.Wpf.Helpers;
+using Winleafs.Models.Models;
+using System.Drawing;
+
+namespace Winleafs.Wpf.Api.Effects.ScreenMirrorEffects
+{
+    public class ScreenMirror : IScreenMirrorEffect
+    {
+        private readonly INanoleafClient _nanoleafClient;
+        private readonly IExternalControlEndpoint _externalControlEndpoint;
+        private readonly List<DrawablePanel> _panels;
+        private readonly int _rectangleSize;
+        private readonly Rectangle _screenBounds;
+        private readonly Rectangle _capturedBounds; //Simple rectangle that start at 0,0 and has width and height set to the rectangle size
+
+        public ScreenMirror(Device device, Orchestrator orchestrator, INanoleafClient nanoleafClient)
+        {
+            _nanoleafClient = nanoleafClient;
+
+            _externalControlEndpoint = _nanoleafClient.ExternalControlEndpoint;
+
+            _screenBounds = MonitorHelper.GetScreenBounds(device.ScreenMirrorMonitorIndex);
+
+            _panels = orchestrator.PanelLayout.GetScaledTriangles(_screenBounds.Width, _screenBounds.Height);
+
+            //Set the rectangle size to 1/3th of the length of a side of the triangle. TODO: test what size is best
```

Comment:
TODO?

Reply:
Intentional, is part of a task in the project.